src/common/ui/pages/requestLoanPage.js

```
@@ -0,0 +1,33 @@
+import { BasePage } from './basePage';
+import { expect } from '../../helpers/pwHelpers';
+
+export class RequestLoanPage extends BasePage {
+  constructor(page, userId = 0) {
+    super(page, userId);
+    this.url = 'https://parabank.parasoft.com/parabank/requestloan.htm';
+  }
+
+  async open() {
+    await super.open(this.url);
+  }
+
+  async fillLoanRequestForm(data) {
+    await this.step(`Fill loan request form`, async () => {
+      await this.page.fill('input#amount', data.amount.toString());
```

This line calls `data.amount.toString()`. If `data` is not an object or `amount` is undefined (for example if the caller passed positional args rather than an object), this will throw `TypeError: Cannot read properties of undefined`. Consider validating `data` first (e.g. assert that `data && typeof data.amount !== 'undefined'`) or normalize inputs so the method accepts both an object and positional values. The tests currently call the action with positional args which would produce such an error unless the action layer builds an object first. See the request_loan spec for current usage.